app/src/main/java/com/graeberj/taskman/auth/presentation/registration/RegistrationViewModel.kt

```diff
@@ -16,44 +20,41 @@ class RegistrationViewModel @Inject constructor(
     private val repository: AuthRepository,
     private val validateForm: ValidateFormUseCase
 ) : ViewModel() {
+
     private val _state = MutableStateFlow(RegistrationState())
     val state = _state.asStateFlow()
 
+    val navigationEvent = MutableSharedFlow<NavigationEvent>()
+    val publicEvent = navigationEvent.asSharedFlow()
+    // working with chat gpt and this was suggested, I'm going to look into this deeper
```

Comment:
I'd rather work with Channels. SharedFlows are intended to be used when there are multiple subscribers (that's where the name comes from). For such UI events, there's typically only one subscriber in the UI though. Channels however come with an integrated buffer which saves events for example when the app is in the background and receives them when it comes back to the foreground.
While with SharedFlows, such events would be lost, a channel can buffer them and keep them. It looks like this in practice:
```
private val eventChannel = Channel<NavigationEvent>()
val navigationEvents = eventChannel.receiveAsFlow() // <- this will be collected on the UI

// ...

viewModelScope.launch {
    eventChannel.send(<your event>)
}
```